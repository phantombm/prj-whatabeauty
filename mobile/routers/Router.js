import React, { Component } from 'react';
import { Router, Scene } from 'react-native-router-flux';

import Main from '../scenes/Main';
import Links from '../scenes/Links';
import Tutorial from '../scenes/Tutorial';

export default class _Router extends Component {
  render() {
    return (
      <Router>
        <Scene key="root">
          <Scene
            key="tutorialRouter"
            initial
            hideNavBar
          >
            <Scene
              key="tutorial"
              component={ Tutorial }
              title="tutorial"
            />
          </Scene>
          <Scene key="mainRouter" hideNavBar>
            <Scene
              key="main"
              component={ Main }
              title="main"
              panHandlers={ null }
            />
            <Scene
              key="links"
              component={ Links }
              title="Links"
              panHandlers={ null }
            />
          </Scene>
        </Scene>
      </Router>
    );
  }
}
