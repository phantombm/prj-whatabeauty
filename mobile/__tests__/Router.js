import React from 'react';
import 'react-native';
import testRenderer from 'react-test-renderer';

import Router from '../routers/Router';

test('renders correctly', () => {
  const tree = testRenderer.create(
    <Router />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
