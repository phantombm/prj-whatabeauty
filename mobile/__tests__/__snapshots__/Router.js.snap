// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      pointerEvents="auto"
      style={
        Object {
          "backgroundColor": "white",
          "bottom": 0,
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "flex": 1,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            pointerEvents="auto"
            style={
              Object {
                "backgroundColor": "white",
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                ],
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                  },
                  undefined,
                ]
              }
            >
              <View>
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "position": "relative",
                      },
                      Object {
                        "height": 1334,
                        "width": 750,
                      },
                    ]
                  }
                >
                  <RCTScrollView
                    automaticallyAdjustContentInsets={false}
                    autoplay={false}
                    autoplayDirection={true}
                    autoplayTimeout={2.5}
                    bounces={false}
                    contentContainerStyle={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                        },
                        undefined,
                      ]
                    }
                    contentOffset={
                      Object {
                        "x": 0,
                      }
                    }
                    horizontal={true}
                    index={0}
                    loop={false}
                    onMomentumScrollEnd={[Function]}
                    onScroll={[Function]}
                    onScrollBeginDrag={[Function]}
                    pagingEnabled={true}
                    removeClippedSubviews={true}
                    renderPagination={[Function]}
                    scrollEventThrottle={16}
                    scrollsToTop={false}
                    showsButtons={false}
                    showsHorizontalScrollIndicator={false}
                    showsPagination={true}
                    showsVerticalScrollIndicator={false}
                  >
                    <View>
                      <View
                        style={
                          Array [
                            Object {
                              "height": 1334,
                              "width": 750,
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                          ]
                        }
                      >
                        <View
                          showsPagination={false}
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#9DD6EB",
                                "flex": 1,
                                "justifyContent": "center",
                                "padding": 15,
                              },
                              Object {
                                "backgroundColor": "#999999",
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "flex": 0.5,
                                "justifyContent": "center",
                                "opacity": 1,
                                "transform": Array [
                                  Object {
                                    "translateX": 0,
                                  },
                                ],
                              }
                            }
                          >
                            <Image
                              source={1}
                              style={undefined}
                            />
                          </View>
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "flex": 0.5,
                                "padding": 30,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "opacity": 1,
                                  "transform": Array [
                                    Object {
                                      "translateX": 0,
                                    },
                                  ],
                                }
                              }
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={true}
                                disabled={false}
                                ellipsizeMode="tail"
                                style={
                                  Array [
                                    Object {
                                      "color": "#fff",
                                      "fontSize": 30,
                                      "paddingBottom": 20,
                                    },
                                    Object {
                                      "color": undefined,
                                    },
                                  ]
                                }
                              >
                                page 1
                              </Text>
                            </View>
                            <View
                              style={
                                Object {
                                  "opacity": 1,
                                  "transform": Array [
                                    Object {
                                      "translateX": 0,
                                    },
                                  ],
                                }
                              }
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={true}
                                disabled={false}
                                ellipsizeMode="tail"
                                style={
                                  Array [
                                    Object {
                                      "color": "#fff",
                                      "fontSize": 20,
                                    },
                                    Object {
                                      "color": undefined,
                                    },
                                  ]
                                }
                              >
                                description 1
                              </Text>
                            </View>
                          </View>
                        </View>
                      </View>
                      <View
                        style={
                          Array [
                            Object {
                              "height": 1334,
                              "width": 750,
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                          ]
                        }
                      >
                        <View
                          showsPagination={false}
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#9DD6EB",
                                "flex": 1,
                                "justifyContent": "center",
                                "padding": 15,
                              },
                              Object {
                                "backgroundColor": "#999999",
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "flex": 0.5,
                                "justifyContent": "center",
                                "opacity": 0,
                                "transform": Array [
                                  Object {
                                    "translateX": 150,
                                  },
                                ],
                              }
                            }
                          >
                            <Image
                              source={1}
                              style={undefined}
                            />
                          </View>
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "flex": 0.5,
                                "padding": 30,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "opacity": 0,
                                  "transform": Array [
                                    Object {
                                      "translateX": 250,
                                    },
                                  ],
                                }
                              }
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={true}
                                disabled={false}
                                ellipsizeMode="tail"
                                style={
                                  Array [
                                    Object {
                                      "color": "#fff",
                                      "fontSize": 30,
                                      "paddingBottom": 20,
                                    },
                                    Object {
                                      "color": undefined,
                                    },
                                  ]
                                }
                              >
                                page 2
                              </Text>
                            </View>
                            <View
                              style={
                                Object {
                                  "opacity": 0,
                                  "transform": Array [
                                    Object {
                                      "translateX": 100,
                                    },
                                  ],
                                }
                              }
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={true}
                                disabled={false}
                                ellipsizeMode="tail"
                                style={
                                  Array [
                                    Object {
                                      "color": "#fff",
                                      "fontSize": 20,
                                    },
                                    Object {
                                      "color": undefined,
                                    },
                                  ]
                                }
                              >
                                description 2
                              </Text>
                            </View>
                          </View>
                        </View>
                      </View>
                      <View
                        style={
                          Array [
                            Object {
                              "height": 1334,
                              "width": 750,
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                          ]
                        }
                      >
                        <View
                          showsPagination={false}
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#9DD6EB",
                                "flex": 1,
                                "justifyContent": "center",
                                "padding": 15,
                              },
                              Object {
                                "backgroundColor": "#999999",
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "flex": 0.5,
                                "justifyContent": "center",
                                "opacity": -1,
                                "transform": Array [
                                  Object {
                                    "translateX": 300,
                                  },
                                ],
                              }
                            }
                          >
                            <Image
                              source={1}
                              style={undefined}
                            />
                          </View>
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "flex": 0.5,
                                "padding": 30,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "opacity": -1,
                                  "transform": Array [
                                    Object {
                                      "translateX": 500,
                                    },
                                  ],
                                }
                              }
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={true}
                                disabled={false}
                                ellipsizeMode="tail"
                                style={
                                  Array [
                                    Object {
                                      "color": "#fff",
                                      "fontSize": 30,
                                      "paddingBottom": 20,
                                    },
                                    Object {
                                      "color": undefined,
                                    },
                                  ]
                                }
                              >
                                page 3
                              </Text>
                            </View>
                            <View
                              style={
                                Object {
                                  "opacity": -1,
                                  "transform": Array [
                                    Object {
                                      "translateX": 200,
                                    },
                                  ],
                                }
                              }
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={true}
                                disabled={false}
                                ellipsizeMode="tail"
                                style={
                                  Array [
                                    Object {
                                      "color": "#fff",
                                      "fontSize": 20,
                                    },
                                    Object {
                                      "color": undefined,
                                    },
                                  ]
                                }
                              >
                                description 3
                              </Text>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </RCTScrollView>
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "bottom": 25,
                          "flex": 1,
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "flex": 0.2,
                          "height": 50,
                          "justifyContent": "center",
                          "opacity": 1,
                          "transform": Array [
                            Object {
                              "translateX": 15,
                            },
                          ],
                        }
                      }
                    >
                      <View
                        accessibilityComponentType={undefined}
                        accessibilityLabel={undefined}
                        accessibilityTraits={undefined}
                        accessible={true}
                        hitSlop={undefined}
                        isTVSelectable={true}
                        nativeID={undefined}
                        onLayout={undefined}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "height": 80,
                            "justifyContent": "center",
                            "opacity": 1,
                            "width": 100,
                          }
                        }
                        testID={undefined}
                        tvParallaxProperties={undefined}
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          disabled={false}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "#fff",
                                "fontSize": 22,
                                "fontWeight": "bold",
                              },
                              Object {
                                "color": "#fff",
                              },
                            ]
                          }
                        >
                          Skip
                        </Text>
                      </View>
                    </View>
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "rgba(255,255,255,.3)",
                            "borderRadius": 7,
                            "height": 13,
                            "marginBottom": 7,
                            "marginLeft": 7,
                            "marginRight": 7,
                            "marginTop": 7,
                            "width": 13,
                          },
                          Object {
                            "backgroundColor": "#fff",
                          },
                          Object {
                            "backgroundColor": "#fff",
                          },
                        ]
                      }
                    />
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "rgba(255,255,255,.3)",
                            "borderRadius": 7,
                            "height": 13,
                            "marginBottom": 7,
                            "marginLeft": 7,
                            "marginRight": 7,
                            "marginTop": 7,
                            "width": 13,
                          },
                          Object {
                            "backgroundColor": "rgba(255,255,255,.3)",
                          },
                        ]
                      }
                    />
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "rgba(255,255,255,.3)",
                            "borderRadius": 7,
                            "height": 13,
                            "marginBottom": 7,
                            "marginLeft": 7,
                            "marginRight": 7,
                            "marginTop": 7,
                            "width": 13,
                          },
                          Object {
                            "backgroundColor": "rgba(255,255,255,.3)",
                          },
                        ]
                      }
                    />
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "flex": 0.2,
                          "height": 50,
                          "justifyContent": "center",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "height": 0,
                            "justifyContent": "center",
                            "opacity": 0,
                            "transform": Array [
                              Object {
                                "translateX": 20,
                              },
                            ],
                            "width": 100,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "height": 80,
                              "justifyContent": "center",
                              "width": 100,
                            }
                          }
                        >
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            disabled={false}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "color": "#fff",
                                  "fontSize": 22,
                                  "fontWeight": "bold",
                                },
                                Object {
                                  "color": "#fff",
                                  "paddingRight": 30,
                                },
                              ]
                            }
                          >
                            Done
                          </Text>
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "height": 0,
                            "justifyContent": "center",
                            "opacity": 1,
                            "width": 100,
                          }
                        }
                      >
                        <View
                          accessibilityComponentType={undefined}
                          accessibilityLabel={undefined}
                          accessibilityTraits={undefined}
                          accessible={true}
                          hitSlop={undefined}
                          isTVSelectable={true}
                          nativeID={undefined}
                          onLayout={undefined}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "height": 80,
                              "justifyContent": "center",
                              "opacity": 1,
                              "width": 100,
                            }
                          }
                          testID={undefined}
                          tvParallaxProperties={undefined}
                        >
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            disabled={false}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "fontFamily": "Arial",
                                  "fontSize": 25,
                                  "fontWeight": "bold",
                                },
                                Object {
                                  "color": "#fff",
                                },
                              ]
                            }
                          >
                            ›
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
